### ISSUE TITLE
Fibaro FGWPF-102 wallplug generates too much traffic

### ISSUE THREAD

**Issue by bm480** [athombv/com.fibaro#204]
I use the wallplug in combination with a waterboiler to detect its usage and to monitor energy consumption. However, using the Z-Wave tools on developer.athom.com I can see that the wallplug generates a lot of traffic coming from Homey itself.

![knipsel](https://user-images.githubusercontent.com/23471238/35966037-fa9086fe-0cbc-11e8-9f75-aca985381cde.PNG)

Also, the Z-Wave log is showing the following:

:
2018-02-08T09:38:01.459Z | ProcessSendData[39626]: To node: 80 with data: 0x70043d0109 and txOptions: AUTO_ROUTE,ACK,EXPLORE
-- | --
2018-02-08T09:38:01.714Z | Command[39628] end: sendData
2018-02-08T09:38:01.714Z | Command[39629] start: sendData
2018-02-08T09:38:01.714Z | ProcessSendData[39627]: To node: 80 with data: 0x70043d0109 and txOptions: AUTO_ROUTE,ACK,EXPLORE
2018-02-08T09:38:01.760Z | Command[39629] end: sendData
2018-02-08T09:38:01.760Z | Command[39630] start: sendData
2018-02-08T09:38:01.761Z | ProcessSendData[39628]: To node: 80 with data: 0x70043d0109 and txOptions: AUTO_ROUTE,ACK,EXPLORE
2018-02-08T09:38:02.097Z | Command[39630] end: sendData
2018-02-08T09:38:02.097Z | Command[39631] start: sendData
2018-02-08T09:38:02.098Z | ProcessSendData[39629]: To node: 80 with data: 0x70043d0109 and txOptions: AUTO_ROUTE,ACK,EXPLORE
2018-02-08T09:38:02.377Z | Command[39631] end: sendData
2018-02-08T09:38:02.378Z | Command[39632] start: sendData
:

Also note that the device is within 3 meters of Homey itself and I have Homey restarted twice a day automatically.

**Comment by caseda** [comment]
Do you have a flow that sends the led ring "on" color? As that is what it is trying to send constantly

**Comment by caseda** [comment]
Ah I see, the parameter number is different for the zwave plus version

**Comment by bm480** [comment]
Yes, that is correct. Has this been changed recently?

**Comment by caseda** [comment]
Mmm the parameter number is right in the driver, are you sure it is the 102? as the parameter it is sending is from the wrong driver

**Comment by caseda** [comment]
And also it is not flagged as secure, that most likely is the cause.
If you are really sure it is the zwave plus version reinclude it

**Comment by bm480** [comment]
On the invoice the article is shown as:
Tussenstekker Schuko Z-wave Plus Fibaro
Artikelnummer: FIB_FGWPF-102-ZW5

Also, the node properties in Homey are as follows:

![knipsel](https://user-images.githubusercontent.com/23471238/35967916-a6358f68-0cc2-11e8-9b6d-6f2bf83c5aa5.PNG)

**Comment by caseda** [comment]
Then you were either lied to, or were send the wrong device.
As you have the non zwave plus version.

**Comment by bm480** [comment]
I'll check the label on the device itself later (I'm not at home right now). I have 3 Fibaro wallplugs in total, ordered at different moments.

Let's assume this is an unsecure device, is it then a bug in the Fibaro app? How can this behaviour be prevented in the future. Could this be the cause of Homey locking up after a few hours?

**Comment by caseda** [comment]
It is an unsecure device, only the plus version has a secure connection (harder to catch the signal and make any sense out of its containing data).

And yes I am 100% sure this is the "older" non zwave plus version.

As the signal is hopping via another device it is entirely possible that that device is the cause of the signal not reaching.
Does that device it is going through working properly?

**Comment by bm480** [comment]
So you're saying that Z-Wave plus devices can only be added as a secure device? I was assuming that if pairing in a secure way does not succeed, it may fall back to pairing as an unsecure device.

Anyway, I will also look at your suggestion that the other device might be causing issues. But repairing the wallplug should reset the neighbouring tables, so traffic will no longer be routed through the other device. I will let you know what happens later this evening.

**Comment by caseda** [comment]
Currently secure zwave devices will always include secure unless the device manufacturer implemented a secure and unsecure inclusion way, fibaro does not have this, so a secure device will always include secure (or fail)

**Comment by bm480** [comment]
Currently, I can see the same behaviour with my washing machine which is also connected to a Fibaro wallplug. The difference being that this one is added as a secure device.

![knipsel](https://user-images.githubusercontent.com/23471238/35977484-83f28278-0ce3-11e8-9adf-0166357aeb2d.PNG)

**Comment by bm480** [comment]
I checked the model number of the non-secure device, which reads: FWGPE-F-102. According to the following website this device is a Z-Wave plus device:

https://products.z-wavealliance.org/products/1653

Why it doesn't pair in secure mode, I have no idea. I re-paired the device today but it still does not go in secure mode.

**Comment by bm480** [comment]
So I updated my flows and for the first time in weeks this integration works again like a charm. So I was happy at first, until I noticed the transmit count rapidly going up again. Z-Wave logging is showing a lot of:

2018-02-09T18:59:30.759Z | Node[180]: [COMMAND_CLASS_CONFIGURATION] [CONFIGURATION_SET] {"type":"Buffer","data":[61,1,9]}
-- | --
2018-02-09T18:59:30.760Z | Node[180]: sendData to COMMAND_CLASS_CONFIGURATION, params 0x043d0109
2018-02-09T18:59:30.760Z | Command[3519] start: sendData
2018-02-09T18:59:30.760Z | ProcessSendData[3517]: To node: 180 with data: 0x70043d0109 and txOptions: AUTO_ROUTE,ACK,EXPLORE
2018-02-09T18:59:30.819Z | Command[3519] end: sendData
:

**Comment by sjaakie33** [comment]
below the log in your message is there any transmit un-acknowledgeor send failure

**Comment by bm480** [comment]
No, Homey just keeps sending data. About 2-5 messages per second.

**Comment by sjaakie33** [comment]
looks like it's out of reach to send and receive configuration
bring homey closer or the device.
sending configuration's to be done as close as possible
pull the plug will solve the sending. 
but then the change of settings will not been done

**Comment by bm480** [comment]
Its not out of reach, it has been at the same location (about 3 meters from Homey in the same room) and it has worked before (until a few weeks ago) without problems. The logging is about changing the LED color as @caseda already mentioned. This apparently requires a configuration change.

Indeed restarting Homey gets rid of the endless sending loop, but I'm looking for a solution that prevents this from happening in the first place... I could do without changing the LED color, but that halves the fun of the integration which is why I just disabled all related flows waiting for a fix.

**Comment by bm480** [comment]
Just updated to Homey 1.5.6 rc11, which feels more stable Z-Wave wise. However, I can still see the same behaviour in which Homey keeps sending messages to the wallplug after the LED has changed color from a flow.

**Comment by bm480** [comment]
Updated to Homey 1.5.8. However, the issue with the Fibaro wall plug remains. After changing the LED-color using a flow Homey keeps sending data. This behaviour can be seen with both the secure and unsecure variants of the device.
